src/login.tsx

```diff
@@ -41,24 +82,87 @@ const ButtonText = styled.Text`
 `;
 
 const Login = () => {
-  return (
-    <Background>
-      <MainText>Bem-vind@ à Taqtile!</MainText>
-      <FieldsCell>
-        <View>
-          <SubText>Email</SubText>
-          <TextBox placeholder={'exemple@exemple.com'} />
-        </View>
-        <View>
-          <SubText>Senha</SubText>
-          <TextBox secureTextEntry={true} />
-        </View>
-        <LoginButton>
-          <ButtonText>Entrar</ButtonText>
-        </LoginButton>
-      </FieldsCell>
-    </Background>
-  );
+
+	const [email, setEmail] = useState<string>('');
+	const [password, setPassword] = useState<string>('');
+	const [emailError, setEmailError] = useState<FieldErrors>();
+	const [passwordError, setPasswordError] = useState<FieldErrors>();
+
+	function containsAnyLetter(str: string) {
+		return /[a-zA-Z]/.test(str);
+	}
+	function containsAnyDigit(str: string) {
+		return /[1-9]/.test(str);
+	}
+
+	const validateElmail = () => {
+		if (email.length === 0){
+			setEmailError('empty');
+		} else if (email.search('@') === -1){
+			setEmailError('structure');
+		} else {
+			var emailSplited = email.split('@',2);
+			if (!emailSplited[1].endsWith('.com') || emailSplited[1].length <= 4 || emailSplited[0].length === 0 ){
+				setEmailError('structure');
+			} else {
+				setEmailError(undefined);
+			}
+		}
+	};
+
+	const validatePassword = () => {
+		if (password.length === 0){
+			setPasswordError('empty');
+		} else if (password.length < 7){
+			setPasswordError('length');
+		} else if (containsAnyLetter(password) && containsAnyDigit(password)){
+			setPasswordError(undefined);
+		} else {
+			setPasswordError('structure');
+		}
+	};
+
+	const validateLogin = () => {
+		validateElmail();
+		validatePassword();
+	};
+
+	// this.render () {
+	return (
+		<Background>
+			<MainText>Bem-vind@ à Taqtile!</MainText>
+			<FieldsCell>
+				<View>
+					<SubText>Email</SubText>
+					<TextBox
+						placeholder={'exemple@exemple.com'}
+						onChangeText={(value) => setEmail(value)}
+						defaultvalue={email}
+						status={emailError}
+					/>
+					{emailError === 'empty' ? <ErrorText>Este campo é obrigatório</ErrorText> :
+						emailError === 'structure' ? <ErrorText>Email inválido</ErrorText> : <ErrorText> </ErrorText>
+					}
+				</View>
+				<View>
+					<SubText>Senha</SubText>
+					<TextBox
+						secureTextEntry={true}
+						onChangeText={(value) => setPassword(value)}
+						defaultvalue={password}
+						status={passwordError}
+					/>
+					{passwordError === 'empty' ? <ErrorText>Este campo é obrigatório</ErrorText> :
+						passwordError === 'structure' ? <ErrorText>Senha deve conter letras e números</ErrorText> :
+						passwordError === 'length' ? <ErrorText>Senha deve ter pelo menos 7 caracteres</ErrorText> : <ErrorText> </ErrorText>
+					}
```

Comment:
Aqui os ternários, dependendo do formatador e do tamanho máximo de caracteres por linha, podem quebrar linhas e acabar dificultando a leitura. Eu reocmendaria, se for o caos, extrair o resultado para constantes antes do retorno pra ajudar nessa leitura.